base/broadcast.jl

```diff
@@ -283,6 +288,17 @@ function broadcast_c(f, ::Type{Tuple}, As...)
     n = length(shape[1])
     return ntuple(k->f((_broadcast_getindex(A, k) for A in As)...), n)
 end
+@inline function broadcast_c(f, ::Type{Nullable}, a...)
+    nullity = all(hasvalue, a)
+    S = promote_eltype_op(f, a...)
+    if isleaftype(S) && null_safe_eltype_op(f, a...)
+        Nullable{S}(f(map(unsafe_get, a)...), nullity)
```

Comment:
Really cool. I guess that didn't work in my other PR because of the intermediate anonymous function.

Yet, I guess from your code that `null_safe_eltype_op` won't be inferred properly unless you define a method for as many arguments as needed? Maybe a generated function would be warranted there at least?